include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,485 @@
+            anchors.erase(it);
+            anchors.insert(gap);
+        }
+        // post-processing: forward update of succeeding gaps
+        update(pos1, pos2-pos1);
+        return true;
+    }
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return iterator{*this};
+    }
+
+    //!\copydoc begin()
+    const_iterator begin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    //!\copydoc begin()
+    const_iterator cbegin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the decorator.
+     * \returns Iterator to the behind last element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator end() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the i-th element as a reference.
+     * \param i The element to retrieve.
+     * \throws std::out_of_range If you access an element behind the last.
+     * \returns A reference of the gapped alphabet type .
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * Strong exception guarantee (never modifies data)..
+     */
+    reference at(size_type const i)
+    {
+        //TODO add SEQAN_UNLIKELY
+        if (i >= size())
+            throw std::out_of_range{"Trying to access element behind the last in gap_decorator."};
+        return (*this)[i];
+    }
+
+    //!\copydoc at()
+    const_reference at(size_type const i) const
+    {
+        //TODO add SEQAN_UNLIKELY
+        if (i >= size())
+            throw std::out_of_range{"Trying to access element behind the last in gap_decorator."};
+        return (*this)[i];
+    }
+
+    //!\copydoc at()
+    constexpr reference operator[](size_type const i) const
+    {
+        assert(i < size());
+        // case 1: no gaps
+        if (!anchors.size()) return value_type((*sequence)[i]);
+        // case 2: gaps
+        anchor_set_iterator it = anchors.lower_bound(gap_t{i, _});
+        if (anchors.size() && it == anchors.end())
```

Comment:
```suggestion
        if (it == anchors.end())
```
you alread know that `anchors.size()` is true because you tested that in line 386